packages/components/inputs/localized-text-input/src/localized-text-input.tsx

```diff
@@ -191,6 +303,7 @@ const LocalizedTextInput = (props) => {
         {shouldRenderLanguagesButton && (
           <LocalizedInputToggle
             isOpen={areLanguagesExpanded}
+            // @ts-ignore
             onClick={toggleLanguages}
```

Comment:
Not sure about this one but we're providing a function (`toggleLanguages`) which has this signature:
```
toggleLanguages(forceIsToggled?: boolean) => void
```
to a prop which has this signature:
```
onClick: (
    event: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLButtonElement>
  ) => void;
```

When dealing with this situations I think is is not completely safe to use functions with different signatures since they can stop working as expected in the future.
Let's imagine the `LocalizedInputToggle.onClick` changes its implementation (yes, I know it's not very probable in this context) and its first param becomes a _boolean_ (`true`): the languages toggle would will ever be `true` so no toggle anymore.

I believe is safer to use an intermediate function in this situations that is tailored for its use case:
```
const onLocalizedInputToggle = () => toggleLanguages();
```

Maybe this is complicate things more than needed, but I wanted to bring this thought up.

Reply:
Exactly, I actually thought about this and felt we should have a talk about this particular prop as it appears to be a bit confusing  on how to handle it.